test-case/check-audio-equalizer.sh

```diff
@@ -59,10 +65,15 @@ func_test_eq()
     }
 
     dlogc "$cmd -D $dev -f $fmt -c $channel -r $rate -d $duration $dummy_file"
-    $cmd -D "$dev" -f "$fmt" -c "$channel" -r "$rate" -d "$duration" "$dummy_file" || {
-        dloge "Equalizer test failure with $conf"
-        return 1
-    }
+    $cmd -D "$dev" -f "$fmt" -c "$channel" -r "$rate" -d "$duration" "$dummy_file" || ret=$?
+
+    [ -r "$eq_last_conf" ] ||
+        die "Failed to read equalizer config from $eq_last_conf"
+
+    sof-ctl -Dhw:"$sofcard" -c name="$double_quoted_id" -s "$eq_last_conf" || die "Failed to restore equalizer config from $eq_last_conf"
+
+    [ "$ret" -eq 0 ] || die "Equalizer test failure with $conf"
```

Comment:
```suggestion
    $cmd -D "$dev" -f "$fmt" -c "$channel" -r "$rate" -d "$duration" "$dummy_file" || ret=$?

    [ -r "$eq_last_conf" ] ||
        die "Failed to read equalizer config from $eq_last_conf"

    sof-ctl -Dhw:"$sofcard" -c name="$double_quoted_id" -s "$eq_last_conf" || die "Failed to restore equalizer config from $eq_last_conf"

    [ "$ret" -eq 0 ] || return 1
```
to keep the current expectation at the caller function.